x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java

```diff
@@ -245,6 +242,24 @@ public void testSettingFilter() throws Exception {
         assertThat(filter, hasItem("transport.profiles.*.xpack.security.*"));
     }
 
+    public void testOnIndexModuleIsNoOpWithSecurityDisabled() throws Exception {
+        Settings settings = Settings.builder()
+                .put(XPackSettings.SECURITY_ENABLED.getKey(), false)
+                .put("path.home", createTempDir())
+                .build();
+        createComponents(settings);
+        IndexSettings indexSettings = IndexSettingsModule.newIndexSettings("foo", Settings.EMPTY);
+        ThreadPool threadPool = mock(ThreadPool.class);
+        when(threadPool.getThreadContext()).thenReturn(threadContext);
+        AnalysisRegistry emptyAnalysisRegistry = new AnalysisRegistry(TestEnvironment.newEnvironment(settings), emptyMap(), emptyMap(),
+            emptyMap(), emptyMap(), emptyMap(), emptyMap(), emptyMap(), emptyMap(), emptyMap());
+        IndexModule indexModule = new IndexModule(indexSettings, emptyAnalysisRegistry, new InternalEngineFactory(), Collections.emptyMap(),
+            () -> true, TestIndexNameExpressionResolver.newInstance(threadPool.getThreadContext()), Collections.emptyMap());
+        security.onIndexModule(indexModule);
+        // indexReaderWrapper is a SetOnce so if Security#onIndexModule had already set an ReaderWrapper we would get an exception here
+        indexModule.setReaderWrapper(null);
```

Comment:
to be pedantic, I would mock the `IndexModule` and use `Mockito#verififyZeroInteractions` .

Reply:
was my first approach but `IndexModule` is a final class.

Reply:
mmkay